src/libsyntax/ext/base.rs

```diff
@@ -776,6 +777,22 @@ impl<'a> ExtCtxt<'a> {
     pub fn name_of(&self, st: &str) -> ast::Name {
         token::intern(st)
     }
+
+    pub fn suggest_macro_name(&mut self, name: &str, span: Span) {
+        use std::cmp::max;
+
+        let mut min: Option<(Name, usize)> = None;
+        let max_dist = max(name.len() / 3, 1);
```

Comment:
Can we avoid duplicating this heuristic everywhere? It was tricky (#26087) and we may want to update it. I don't want to block on this, but I'd block on having FIXME comment with filed issue or at least having cross reference to same heuristic in resolve, both here and in resolve. (Best to just fix it though.)